### PR TITLE
Add Microstates example

### PR DESCRIPTION
**Pull request by Robdel12**
First off I love this idea! Thank you for putting this together. 🎉 

## What is this?

This PR adds Microstates to the examples list for React Web. We almost have the React Native example working but there's something funky going on with the provider so I'm going to pull that in once we figure out what's going on there 🤔 

Microstates are a typed, composable, and immutable state container. If you look at `list.js` you can see that it's just a JS class that points properties to types. For example `allItems` is an array of strings and `newItemText` is just a string. The rest of the `list` class defines the state transitions.

### CODE REVIEW

**Comment by GantMan** [comment]
I'll review this today!

File: React/microstates/src/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    return (
+      <State
+        type={List}
+        value={{ allItems: ["nachos", "burritos", "hot dog"] }}
+      >
+        {list => (
+          <div style={styles}>
+            <h2>Welcome to Microstates</h2>
+            <ListItems allItems={list.state.allItems} />
+            <AddItems
```
Comment (GantMan) [inline React/microstates/src/index.js]: Can you please switch these two?  The AddItems is usually above the ListItems in all other examples.

image

Comment (Robdel12) [reply React/microstates/src/index.js]: Ha! I didn't even realize I flipped them around. I'll get a change pushed up for sure

**Comment by Robdel12** [comment]
@GantMan I'd love to pair up on it if you'd like! The issue is `list` from the provider is `null` which causes `list.state.newItemText` to blow up. There's something funky going on inside of here but I'm not _quite_ sure what it is yet 🤔 https://github.com/microstates/react/blob/master/src/microstates.js

**Comment by GantMan** [comment]
Is this what you're getting on native?

```
bundling failed: Error: Unable to resolve module `@microstates/react` from `/Users/gantman/Documents/Projects/training/ReactStateMuseum/ReactNative/Microstates/App.js`: Module `@microstates/react` does not exist in the Haste module map
```

**Comment by GantMan** [comment]
It looks to me from a cursory examination that microstates has at it's core a dependency on defining the Constructor prototype during creation.  This calls `Object.defineProperty` in microstates, and then KABOOM.  Because the polyfill on Native is not configurable.

Microstates dist line 30:  `Object.defineProperty(target, descriptor.key, descriptor);`

Without this microstates.create is not a function.  Not sure the workaround, but it's a doozy.

If you don't figure it out, we can merge microstates web, and put a note that it simply doesn't work with Native.  I think that's valueable info for people out there.  Maybe even file a ticket with microstates letting them know?

**Comment by Robdel12** [comment]
That sounds good! We’re working on figuring out something for React Native — it’s absolutely something we want to support! I’ll circle back 😃

**Pull request merged**